feat: fixed session duration + better visibility of changes done via `bench browse`

akhilnarang opened the pull request:
- feat: allow created a session for a fixed duration via `bench browse`
- refactor: fix typo in `impersonated`
- feat(browse): allow passing a user for `impersonated_by`

<hr>

Based on our requirements of basically not unconditionally allowing "Login as Administrator" through Frappe Cloud, we need to allow setting a customized session duration, and maintaining a better audit trail, showing who exactly made the edits.

akhilnarang committed:
feat: allow created a session for a fixed duration via `bench browse`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
refactor: fix typo in `impersonated`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
feat(browse): allow passing a user for `impersonated_by`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

ankush commented on frappe/commands/site.py:
```diff
@@ -1192,6 +1200,9 @@ def browse(context: CliCtxObj, site, user=None):
 	frappe.connect()
 
+	frappe.flags.session_duration = duration
+	frappe.flags.audit_user = user_for_audit
```
If possible, avoid these flags and directly use them on `login_manager.login` or `login_manager.impersonate`. 

`impersonate` is a better candidate since that's used from UI too.

ankush commented:
Weird edge case: Double impersonation :smile: 

Support agents do want to impersonate some specific user when they are debugging permission issue. (original intent of the feature)

akhilnarang commented:
> Weird edge case: Double impersonation 😄
> 
> Support agents do want to impersonate some specific user when they are debugging permission issue. (original intent of the feature)

Hmm yeah missed this, will just implement something similar to impersonate separately then so that both can work simultaneously.

akhilnarang committed:
refactor: don't use impersonate directly, use similar logic

This will allow impersonating as well

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang commented:
![image](https://github.com/user-attachments/assets/745ea24e-6a35-4d06-b47c-5305dbfcbc21)

Timeline would look like this if impersonating now.

akhilnarang committed:
refactor: use an alternate key for handling expiry

This allows for less changes to update() + allows impersonated sessions to not end later

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
